src/components/AllMonstersList.js

```diff
@@ -1,41 +1,52 @@
 import React, { useEffect, useState } from 'react'
 import styled from 'styled-components'
 import saveLocally from './lib/saveLocally'
+import OneMonsterItem from './OneMonsterItem'
 
 export default function AllMonstersList() {
-  const [monsterList, setMonsterList] = useState([])
+  const [monsterDetails, setMonsterDetails] = useState([])
 
   useEffect(() => {
-    const loadedArray = JSON.parse(localStorage.getItem('monsternames'))
+    const locallyLoadedMonsterNames = JSON.parse(
+      localStorage.getItem('monsternames')
+    )
 
-    if (!loadedArray || loadedArray.length === 0) {
+    if (!locallyLoadedMonsterNames || locallyLoadedMonsterNames.length === 0) {
       fetch(`https://www.dnd5eapi.co/api/monsters`)
         .then((res) => res.json())
         .then((data) => {
-          setMonsterList(data.results)
           saveLocally('monsternames', data.results)
+          data.results.forEach((monster) => {
+            fetch(`https://www.dnd5eapi.co${monster.url}`)
+              .then((res) => res.json())
+              .then((data) => {
+                console.log(data.index)
+                setMonsterDetails((monsterDetails) => [...monsterDetails, data])
+              })
+          })
         })
         .catch((err) => console.log(err))
-    } else {
-      setMonsterList(loadedArray)
     }
   }, [])
 
+  if (monsterDetails.length !== 0) {
+    saveLocally('monsterdetails', monsterDetails)
+  } else if (localStorage.getItem('monsterdetails')) {
+    setMonsterDetails(JSON.parse(localStorage.getItem('monsterdetails')))
+  }
+  console.log(monsterDetails)
   return (
     <>
       <HeadlineStyled>Monsters of DnD</HeadlineStyled>
-      {monsterList.map((monster, index) => (
-        <MonsterEntryStyled key={index}>{monster.name}</MonsterEntryStyled>
+      {monsterDetails.map((monster, index) => (
+        <OneMonsterItem key={index} monster={monster}></OneMonsterItem>
```

Comment:
index is not a good practice to use it as a key. You can use in your case the monster name, if unique.

Reply:
Good spot! In my Case the api provides the name of the monster in the index, but without any caps and special signs like - or .
For Ex: 
Name: Aboleth
index: aboleth
So in this case it works, but i made a note of that for the future.